### PR TITLE
MULE-9512: Refactor deployment service tests to avoid using pre-packa…

### PR DESCRIPTION
**Pull request by pablokraan**
…ged artifacts

_ Removing duplication from AbstractFileBuilder and ZipUtils
_ Making static artifact file builders on DeploymentServiceTestCase non static to avoid issues when different builders write to the same file

### CODE REVIEW

File: modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java [hunk 1]
```diff
@@ -146,6 +144,7 @@ public File getArtifactFile() throws Exception
             checkArgument(!StringUtils.isEmpty(fileName), "Filename cannot be empty");
 
             final File tempFile = new File(getTempFolder(), fileName);
+            //assertThat("Artifact already exists. Use a different name: " + fileName, tempFile.exists(), is(false));
```
Comment (dfeist) [inline modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java]: Remove or uncomment.

File: modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java [hunk 2]
```diff
@@ -255,16 +209,8 @@ private void buildBrokenZipFile(File tempFile) throws IOException
...
-
-        final String file;
-        final String alias;
-
-        protected ZipResource(String file, String alias)
-        {
-            this.file = file;
-            this.alias = alias;
-        }
+        return new ArrayList<>();
```
Comment (elrodro83) [inline modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java]: Is there possibilities in the future that this returns something other than an empty list?
If not, why not just make this method abstract and remove the super call form the extensions?

File: modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java [hunk 3]
```diff
@@ -154,7 +153,10 @@ public File getArtifactFile() throws Exception
             }
             else
             {
-                buildZipFile(tempFile);
+
```
Comment (elrodro83) [inline modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java]: extra line

File: modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java [hunk 4]
```diff
@@ -154,7 +153,10 @@ public File getArtifactFile() throws Exception
...
+                final List<ZipResource> zipResources = new LinkedList<>(resources);
+                zipResources.addAll(getCustomResources());
+                ZipUtils.compress(tempFile, zipResources.toArray(new ZipResource[0]));
```
Comment (elrodro83) [inline modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java]: static import

File: modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java [hunk 5]
```diff
@@ -255,16 +206,5 @@ private void buildBrokenZipFile(File tempFile) throws IOException
...
-        final String file;
-        final String alias;
-
-        protected ZipResource(String file, String alias)
-        {
-            this.file = file;
-            this.alias = alias;
-        }
-    }
+    abstract List<ZipResource> getCustomResources() throws Exception;
```
Comment (elrodro83) [inline modules/launcher/src/test/java/org/mule/module/launcher/builder/AbstractArtifactFileBuilder.java]: protected

**Pull request merged**